src/language/walker/programAwareRuleWalker.ts

```diff
@@ -20,6 +20,7 @@ import * as ts from "typescript";
 import { IOptions } from "../rule/rule";
 import { RuleWalker } from "./ruleWalker";
 
+// tslint:disable-next-line:deprecation
 export class ProgramAwareRuleWalker extends RuleWalker {
```

Comment:
Oh, I'd forgotten this exists! 😄

Could you please also mark this as deprecated @Nazanin1369? Its runtime works the same as `RuleWalker` _(but with the type checker and program)_, so it should be deprecated for the same reasons. I'll update the linked issue.

Reply:
Sure

Reply:
scopeAwareRuleWalker.ts already has the deprecated annotation.